### ISSUE TITLE
OpenCV V3.1.0 on Raspberry Pi : namedWindow bus warnings.

### ISSUE THREAD

**Issue by BeaconJohn** [opencv/opencv#6159]
I recently upgraded from V2.4.11 to V3.1.0. A really messy process because I'm using nonfree modules, but that's another story.
Everything will now recompile but I am getting issues at run time.
One section of code that ran correctly under 2.4 has three calls to 'namedWindow()' in series. The first call results in the following warning;
(Dummy:2718): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
The other two calls result in;
(Dummy:2718): WARNING **: Can't load fallback CSS resource: Failed to import: The resource at '/org/gnome/adwaita/gtk-fallback.css' does not exist

'Dummy' is the name of the first window. Note, this name appears in the warnings for all three calls despite them having different names.

Subsequent calls to imShow() further on in the routine cause a segmentation fault exception and the process terminates.
OpenCV is linked with GTK+-3.0 version 3.10.2.
The GUI config for OpenCV 3.1.0 (from the build log file) is;
GUI:
      QT:                        NO
      GTK+ 3.x:              YES (ver 3.10.2)
      GThread  :             YES (ver 2.40.0)
      GtkGlExt :              NO
      OpenGL support :  NO
     VTK support :          NO

I have tried loading GNOME3-dev but that made no difference. 
I have trawled the internet but can find no clue to what is going on. Can anyone help?

**Comment by mshabunin** [comment]
Please, provide minimal reproducing example. GTK warnings are probably not related with the crash ([first](https://bugs.launchpad.net/ubuntu/+source/dbus/+bug/1217757), [second](https://bugzilla.redhat.com/show_bug.cgi?id=1057889)).

**Comment by BeaconJohn** [comment]
Hi Maksim, thanks for responding. I have attached two files;
namedWindow.cpp is a minimum file that will cause the issue. It simply creates a window, loads an image file, shows image in window. In this instance, the image file is passed as a program argument, in the full code the file name is read from a text file.
The trace file is the output from stdout/stderr. I have annotated it so you can see at what point the warnings are issued. There appears to be no difference between free running the program or running it via the debugger other than the debugger traps the exception whereas the free run simply exits after the last message.
I'm using the Netbeans IDE with the target resident tool chain.

Attachments;
[namedWindow.cpp.txt](https://github.com/Itseez/opencv/files/145437/namedWindow.cpp.txt)

[namedWindow trace.pdf](https://github.com/Itseez/opencv/files/145439/namedWindow.trace.pdf)

**Comment by BeaconJohn** [comment]
One added bit of information; It would appear that, contrary to my first impressions, the first call to namedWindow() causes all three warnings, subsequent calls do not. However the final exception is still raised even if I don't use that first window (which is why it's called Dummy)

**Comment by alalek** [comment]
> OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow

I believe real problem is in `imread()` call. It can't read image and returns empty result.

**Comment by BeaconJohn** [comment]
I think it's a GTK issue. I've just tried a simple GTK demo app I wrote some time ago and get exactly the same result. Nothing to do with OpenCV. 

I'll check out the imread() thing though, that may be something completely different. What is worrying me now though that installing OpenCV V3.1.0 appears to have had effects outside of the package. All of this worked before the upgrade (including the GTK demo)

**Comment by mshabunin** [comment]
Please, add the following line after `imread`:
```.cpp
if (img_object.empty()) cout << "Can not read image: " << argv[1] << endl;
```

What method did you use to install OpenCV? Did you build it by yourself on device or on host?

**Comment by BeaconJohn** [comment]
Well spotted. imread() has always worked up to now so I assumed that the zero width & height reported was a throw back from the window problem. 

I'm surprised that imread doesn't throw an exception if it can't find the file. _imageobject_.empty() tells you that there's no image but it doesn't tell you why; file size of zero, can't read the file, file doesn't exist. Potentially important stuff in a full blown application.

This looks like an environment issue which I don't fully understand. If I enter the absolute filepath then it works but, again, until the upgrade all I needed to enter was the file name as the image file is in the same folder as the rest of the project. (Not well structured I know but it's early days). I now need to try to discover what OpenCV believes to be the default file location so I can look at using relative paths.

OpenCV was built on the target machine. It took over 14 hours to compile an I had to try it several time because I struggled to get the 'contrib' modules included in the build. This has all taken so much time and effort, I've almost forgotten what I was trying to achieve in the first place.

I'll keep this thread open for a while I feed what I've learned back into the full application just in case something else jumps out at me. 
I now also now need to go and talk to the GTK world to find out what's wrong with the windows side of things.
Thanks for your help so far,
john.
